Rubberduck.UnitTesting/UnitTesting/TestEngine.cs

```diff
@@ -0,0 +1,305 @@
+                    using (var typeLibWrapper = _wrapperProvider.TypeLibWrapperFromProject(moduleName.ProjectId))
+                    {
+                        try
+                        {
+                            VBEInteraction.RunDeclarations(_typeLibApi, typeLibWrapper, TestDiscovery.FindModuleInitializeMethods(moduleName, _state));
+                        }
+                        catch (COMException ex)
+                        {
+                            Logger.Error(ex, "Unexpected COM exception while initializing tests for module {0}. The module will be skipped.", moduleName.Name);
+                            foreach (var method in moduleTestMethods)
+                            {
+                                OnTestCompleted(method, new TestResult(TestOutcome.Unknown, AssertMessages.TestRunner_ModuleInitializeFailure));
+                            }
+                            continue;
+                        }
+                        foreach (var test in moduleTestMethods)
+                        {
+                            // no need to run setup/teardown for ignored tests
+                            if (test.Declaration.Annotations.Any(a => a.AnnotationType == AnnotationType.IgnoreTest))
+                            {
+                                OnTestCompleted(test, new TestResult(TestOutcome.Ignored));
+                                continue;
+                            }
+
+                            try
+                            {
+                                fakes.StartTest();
+                                try
+                                {
+                                    VBEInteraction.RunDeclarations(_typeLibApi, typeLibWrapper, testInitialize);
+                                }
+                                catch (COMException trace)
+                                {
+                                    OnTestCompleted(test, new TestResult(TestOutcome.Inconclusive, AssertMessages.TestRunner_TestInitializeFailure));
+                                    Logger.Trace(trace, "Unexpected COMException when running TestInitialize");
+                                    continue;
+                                }
+                                var result = RunTestMethod(typeLibWrapper, test);
+                                // we can trigger this event, because cleanup can fail without affecting the result
+                                OnTestCompleted(test, result);
+                                try
+                                {
+                                    VBEInteraction.RunDeclarations(_typeLibApi, typeLibWrapper, testCleanup);
+                                }
+                                catch (COMException cleanupFail)
+                                {
+                                    // Apparently the user doesn't need to know when test results for subsequent tests could be incorrect
```

Comment:
Preserving previous behavior I guess. Suggestions welcome! 😄

Reply:
Why not report it as a separate error row in the UI? E.g. `Test_Foo (Cleanup) | Cleanup encountered an error: blah`?